services/agent/src/simcore_service_agent/volumes_cleanup/_core.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from .._settings import ApplicationSettings
+from ._docker import delete_volume, docker_client, get_dyv_volumes, is_volume_used
+from ._s3 import store_to_s3
+
+logger = logging.getLogger(__name__)
+
+
+async def backup_and_remove_volumes(settings: ApplicationSettings) -> None:
+    async with docker_client() as client:
+        dyv_volumes: list[dict] = await get_dyv_volumes(client)
+
+        if len(dyv_volumes) == 0:
+            return
+
+        cleaned_up_volumes_count = 0
+        logger.info("Beginning cleanup.")
+        for dyv_volume in dyv_volumes:
```

Comment:
so this is actually the main purpose of this service. now my question here is:
- you have a healthcheck that appears to run on a separate server right?
- if that is the case, how will that healthcheck find that the background task failed for whatever reason? is there a connection?

Reply:
So now the health check looks at the status of the `task_monitor`. If this one is running it assumes it's running.

I'd like to connect it with the monitoring infrastructure to make sure that if the tasks raise too many errors it will be unhealthy.